### ISSUE TITLE
update contributing guide  to reflect current code structure and steps to get started with sveltekit dev app

### ISSUE THREAD

**Issue by jschlesser** [nextauthjs/next-auth#6652]
### Environment

  System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1
    Memory: 64.81 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.13.0 - ~/.nvm/versions/node/v18.13.0/bin/node
    npm: 9.3.0 - /opt/homebrew/bin/npm
  Browsers:
    Firefox: 105.0.1
    Safari: 16.2

### Reproduction URL

https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md

### Describe the issue

First, I have added this as a bug because following the instructions to get started didn't work for me, at least trying to get started with the sveltekit dev app.   It's a prelude to getting a developer productive enough to work on fixing a bug in the code vs the docs.

I have worked through 3 issues when getting started contributing, 2 in the contributing guide https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md and try to start the svelte-kit and 1 in the OAuth provider instructions.  https://authjs.dev/getting-started/oauth-tutorial#2-configuring-oauth-provider

1.  When following the 'setting up your local environment' instructions for contributing, the instructions fail to mention that you need to run `pnpm build` after step 3. `pnpm install` otherwise when starting the svelte-kit development environment the developer will encounter the following error:
`Error: Failed to resolve entry for package "@auth/sveltekit". The package may have incorrect main/module/exports specified in its package.json.`

2.  The location of the .env files in step 4 is incorrect.   The appropriate .env files are inside of the dev application specific code.  e.g. `apps/dev/.env.local.example` should be `apps/dev/sveltekit/.env.example` for example.   In this case the names of the files aren't consistent either.

3.  When configuring the OAuth app provider for GitHub for the sveltekit example, the callback url in the docs here  https://authjs.dev/getting-started/oauth-tutorial#2-configuring-oauth-provider was incorrect.   It references a url path of `/api/auth/callback/github`.  I got a 404 error on that, it looks like the framework was expecting `/auth/callback/github`.   When I configured GitHub with `/auth/callback/github` and fixed issues 1 and 2 and started the sveltekit development app, everything works and I'm able to login and logout.

A nice to have would be to instruct the developer to also set up the GitHub OAuth application as well, probably immediately after step 2. in the contributing guide.  

### How to reproduce

Follow the instructions in the contributing guide https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md and try to start the svelte-kit 

### Expected behavior

For issue 1.   When starting the sveltekit app in app/dev/sveltekit the server should start and when the user navigates to the homepage they should get a webpage instead of a stack trace.

For issue 2.  The developer should be directed to the appropriately named example env files in the correct location.

For issue 3.  When the developer clicks on the GitHub button to authenticate the sveltekit dev app, they should be redirected to the GitHub auth page.

**Comment by jschlesser** [comment]
Hmm, it looks like the sveltekit framework is overriding the general default callback url used in the contributing oauth provider docs.

https://github.com/nextauthjs/next-auth/blob/cee1bddbd5727aba9bf53272bef03d71baa63cb3/packages/frameworks-sveltekit/src/lib/index.ts#L263
vs instructions here:
https://authjs.dev/getting-started/oauth-tutorial#2-configuring-oauth-provider

Maybe its best to have a getting started guide for the sveltekit auth framework?

**Comment by balazsorban44** [comment]
As a note, authjs.dev is under development. Parts of the docs has been lifted from the old docs and have not been ported to being framework agnostic in wording.

we are working towards that.

As for working within the repo, we use Turborepo and pnpm with corepack enabled.

Build before dev should not be needed (at least as a manual step), I guess there's a bad dependency graph and the turbo confif needs to be updated maybe

https://github.com/nextauthjs/next-auth/blob/main/turbo.json

**Comment by jschlesser** [comment]
Just leaving a quick note here.   I have a few other issues that I have logged and I have some PR's for them to fix getting everything working in my environment (cloudflare).   I will come back to this issue and fix up the docs for generic sveltekit as well as cloudflare when it feels like everything has settled in my environment and I feel like I have a good handle on everything.
